libraries/lib-wave-track/AudioSegmentSampleView.cpp

```diff
@@ -0,0 +1,34 @@
+#include "AudioSegmentSampleView.h"
+
+size_t
+AudioSegmentSampleView::Copy(float* buffer, size_t bufferSize) const
+{
+   size_t toWrite =
+      std::min(mLength.as_long_long(), static_cast<long long>(bufferSize));
+   size_t written = 0u;
+   size_t offset = mStart;
+   for (const auto& block : mBlockViews)
+   {
+      const auto toWriteFromBlock = std::min(block->size() - offset, toWrite);
+      std::copy(
+         block->data() + offset, block->data() + offset + toWriteFromBlock,
+         buffer + written);
+      toWrite -= toWriteFromBlock;
+      written += toWriteFromBlock;
+      offset = 0;
+   }
+   return written;
+}
+
+sampleCount AudioSegmentSampleView::GetSampleCount() const
+{
+   return mLength;
+}
+
+AudioSegmentSampleView::AudioSegmentSampleView(
+   std::vector<BlockSampleView> blockViews, size_t start, sampleCount length)
+    : mBlockViews(std::move(blockViews))
+    , mStart(start)
+    , mLength(length)
```

Comment:
I prefer to use C++11 universal initialization syntax  '{...}' where I can for member initializers.